include/swift/Syntax/TokenKinds.def

```diff
@@ -77,6 +77,12 @@
 #define PAT_KEYWORD(kw) SWIFT_KEYWORD(kw)
 #endif
 
+/// CONTEXTUAL_KEYWORD(kw)
+///   Keywords that are endemic to specific contexts, i.e. precedence group decls.
+#ifndef CONTEXTUAL_KEYWORD
+#define CONTEXTUAL_KEYWORD(kw) SWIFT_KEYWORD(kw)
+#endif
```

Comment:
hmm, could you avoid changing `TokenKinds.def` file? We currently parse these contextual keywords as identifiers and we have logic to promote them from identifier to keywords. Changing their kinds here can have larger impact than being necessary. Also, we have other contextual keywords too, e.g. `get` and `set`; it's hard to be exhaustive.

Reply:
I thought this would be a nice starting point to clean up all of the string literal checks and conversions from identifiers that contextual keywords are currently suffering. In addition, even the highlighting is broken on them, possible as a consequence. Will this cause `Keyword cannot be used as an identifier here` in unnecessary places?

Reply:
These contextual keywords can be used as regular Swift identifiers, i.e. the code snippet is entirely valid:
```
func foo() {
    let get = 1
    let associativity = 2
    let higherThan = 3
    ...
}
```
However, the identifiers here are keywords in some circumstances. We have two ways to handle this bifurcation: (1) lexing them as identifier and later promote them during parsing to keywords, or (2) lexing them as contextual keywords and later demote them during parsing to identifier. The current implementation is using (1), and the change you made is changing it to (2). I cannot see any apparent benefit of using one over the other. So, that's why I suggest we avoid changing the kinds in `TokenKinds.def` to keep things simple.

Reply:
Ah. I agree, (1) is the most convenient.